lib/Sema/TypeCheckDecl.cpp

```diff
@@ -6393,7 +6393,8 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
         AccessScope desiredAccessScope = AccessScope::getPublic();
         switch (access) {
         case Accessibility::Private:
-          assert(ED->getDeclContext()->isModuleScopeContext() &&
```

Comment:
Yeah, I considered just returning early at the top if the decl is invalid. However, I noticed that it was checked for validity again below, so it's probably intentional that invalid decls might make it here.